tests/unit/test_client.py

```diff
@@ -119,6 +119,14 @@ def setUp(self):
     def tearDown(self):
         self.endpoint_creator_patch.stop()
 
+    def create_mock_emitter(self, responses=None):
+        if responses is None:
+            responses = []
+
+        emitter = mock.Mock()
```

Comment:
Can you add a `spec=` here?

Reply:
Yep.

Reply:
Looks like this causes issues on Python 2, gonna leave the mock without the spec as that's how it was before.

Reply:
Did some more digging on why this was happening. We apply a `copy.copy` on the event_emitter [here](https://github.com/joguSD/botocore/blob/3774b8d4488fa5567cc589cb33df6dd9ad122866/botocore/args.py#L62) which seems to cause issues on Python 2. What appears to happen is the copy becomes an actual instance of the specified class that never had its constructor called (and thus can be missing attributes).